### ISSUE TITLE
Inline completions are sometimes not shown even when they are available

### ISSUE THREAD

**Issue by ayim** [microsoft/vscode#134589]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.60.2
- OS Version: Windows 21H2 22000.194 (Windows 11)

Steps to Reproduce:

- When a multi-token inline completion is shown, open the Intellisense window
- The tokens after the current token are no longer shown when the Intellisense window is open
- Upon closing the Intellisense window, the multi-token inline completion is shown again

**Comment by jinu-jang** [comment]
Thanks for the clarification.
So from my testing with this in mind, if the dropdown Intellisense was triggered on the context `console.`, `selectedCompletionInfo.text` would be `".log"` and `document.getText(selectedCompletionInfo.range)` would be `"."`

In some scenarios, `selectedCompletionInfo.text` would not start with `document.getText(selectedCompletionInfo.range)` and it seems that those cases wouldn't be visualized as ghost text.
So would it be safe for us to check if `selectedCompletionInfo.text` is prefixed by `document.getText(selectedCompletionInfo.range)`, and if so, just remove the prefix part?

**Comment by hediet** [comment]
> In some scenarios, selectedCompletionInfo.text would not start with document.getText(selectedCompletionInfo.range) and it seems that those cases wouldn't be visualized as ghost text.

Yes, in such cases the suggestion is a replacement that changes existing characters in the text buffer.
I agree that it does not make sense to query the inline completion provider in this case, as the inline completion would not be shown anyways.

> So would it be safe for us to check if selectedCompletionInfo.text is prefixed by document.getText(selectedCompletionInfo.range), and if so, just remove the prefix part?

Yes!

**Comment by jinu-jang** [comment]
That was greatly helpful, but the new insights actually led to a more nuanced confusion.
So excuse my full write up of my problem below. Hopefully this can allow us to see eye-to-eye on what my questions are.

================

So there are currently 4 scenarios I'm trying to capture w/ one logic:

1. Automatic trigger on dot:
Context: `"console."`, Intended full line: `"console.log();"`
`context.selectedCompletionInfo.text === ".log"`
`document.getText(context.selectedCompletionInfo.range) === "."`
Need to return `"log();"` which is `context.selectedCompletionInfo.text.substr(".".length) + "();"`

2. Automatic trigger during variable:
Context: `"conso"`, Intended full line: `"console.log();"`
`context.selectedCompletionInfo.text === "console"`
`document.getText(context.selectedCompletionInfo.range) === "conso"`
Need to return `"console.log();"` which is `context.selectedCompletionInfo.text + "();"`

3. Automatic trigger during property
Context: `"console.lo"`, Intended full line: `"console.log();"`
`context.selectedCompletionInfo.text === ".log"`
`document.getText(context.selectedCompletionInfo.range) === ".lo"`
Need to return `"log();"` which is `context.selectedCompletionInfo.text.substr(".".length) + "();"`

4. Automatic trigger after different non-alphanumeric character
Context: `"console.log(messa"`, Intended full line: `"console.log(message);"`
`context.selectedCompletionInfo.text === "message"`
`document.getText(context.selectedCompletionInfo.range) === "messa"`
Need to return `"message);"` which is `context.selectedCompletionInfo.text + ");"`

================

> So would it be safe for us to check if `selectedCompletionInfo.text` is prefixed by `document.getText(selectedCompletionInfo.range)`, and if so, just remove the prefix part?

Notice that with this logic I had previously suggested, this will get Cases 1 and 3 correct,
but end up returning `"le.log();"` on Case 2, and `"ge);"` on Case 4, both of which won't be visualized by the InlineCompletion API.

Also keep in mind that for Cases 1 and 3, if we return `".log();"`, our completion won't be shown, and we need to explicitly remove the `"."` before we're return our results.

With Case 4 showing us that other non-alphanumeric characters such as parenthesis aren't actually a part of `context.selectedCompletionInfo.text`, the only conclusion I can come to is that `"."` is the only character that's treated specially.

Hence my confusion in detail is:
1. How VSCode currently does prefix matching, and why `"."` is treated differently?
2. Whether or not `"."` is the only character that's treated differently.
2. If this special behavior of `"."` is defined by VSCode, or by other language servers.
3. As an extension of 2, can there be other characters we need to account for, or what's a more general way of checking for these sorta characters.
